packages/smart-contracts/test/contracts/BatchErc20Payments.test.ts

```diff
@@ -0,0 +1,400 @@
+import { ethers, network } from 'hardhat';
+import { BigNumber, Signer } from 'ethers';
+import { expect, use } from 'chai';
+import { solidity } from 'ethereum-waffle';
+import { TestERC20__factory, TestERC20, BatchPayments } from '../../src/types';
+import { batchPaymentsArtifact } from '../../src/lib';
+
+use(solidity);
+
+const logInfos = false;
+
+describe('contract: BatchPayments: ERC20', () => {
+  let payeeOne: string;
+  let payeeTwo: string;
+  let payeeThree: string;
+  let payeeFour: string;
+  let feeAddress: string;
+
+  let spenderAddress: string;
+  let spender1Address: string;
+  let spender2Address: string;
+  let tokenAddress: string;
+  let token1Address: string;
+  let batchAddress: string;
+
+  let owner: Signer;
+  let spender: Signer;
+  let spender1: Signer;
+  let spender2: Signer;
+
+  let beforeERC20Balance: BigNumber;
+  let afterERC20Balance: BigNumber;
+  let beforeERC20Balance1: BigNumber;
+  let afterERC20Balance1: BigNumber;
+
+  const referenceExample1 = '0xaaaa';
+  const referenceExample2 = '0xbbbb';
+  const referenceExample3 = '0xcccc';
+  const referenceExample4 = '0xdddd';
+
+  let token: TestERC20;
+  let token1: TestERC20;
+  let batch: BatchPayments;
+
+  const erc20Decimal = BigNumber.from('1000000000000000000');
+
+  let amount = 2;
+  let feeAmount = 1;
+  let nbTxs: number;
+
+  let tx;
+
+  before(async () => {
+    [, payeeOne, payeeTwo, payeeThree, payeeFour, feeAddress] = (await ethers.getSigners()).map(
+      (s) => s.address,
+    );
+    [owner, spender, spender1, spender2] = await ethers.getSigners();
+    batch = batchPaymentsArtifact.connect(network.name, owner);
+    token = await new TestERC20__factory(owner).deploy(erc20Decimal.mul(10000));
+    token1 = await new TestERC20__factory(owner).deploy(erc20Decimal.mul(10000));
+
+    spenderAddress = await spender.getAddress();
+    spender1Address = await spender1.getAddress();
+    spender2Address = await spender2.getAddress();
+    tokenAddress = await token.address;
+    token1Address = await token1.address;
+    batchAddress = await batch.address;
+
+    await token.connect(owner).transfer(spenderAddress, 1000);
+    await token.connect(owner).transfer(spender1Address, 160);
+    await token.connect(owner).transfer(spender2Address, 160);
+    await token.connect(spender).approve(batchAddress, 1000);
+    await token.connect(spender2).approve(batchAddress, 170);
+
+    // 2nd token
+    await token1.connect(owner).transfer(spender2Address, 1000);
+    await token1.connect(spender2).approve(batchAddress, 1000);
+  });
+
+  it('Should execute a batch payments of a ERC20 to four accounts', async function () {
```

Comment:
why do we need this method if there is no payment reference?

Reply:
deleted because no more orphan payment in the contract.